rsconf/component/devbox.py

```diff
@@ -98,23 +95,32 @@ def _gen_secrets(self, jc):
         s = super().gen_identity_and_host_ssh_keys(jc, "host", encrypt_identity=True)
         self.secrets = PKDict({k: s[k] for k in ("host_key_f", "identity_pub_f")})
 
-    def _jupyter(self, z, jupyter_bashrc_path):
+    def _jupyter_bashrc(self, z, path):
+        def _env(name, value):
+            v = str(value)
+            # TODO probably could recursively quote
+            if "'" in v:
+                raise ValueError(
+                    f"single quote in value={v} of bash variable name={name}"
+                )
+            self.rsconf_append(
+                path,
+                f"export {name}='{v}'",
+            )
+
+        self.install_access(mode="600")
```

Comment:
How do you decide these? Just least privilege possible? I looked on my machine and saw 644 so I went with it...

Reply:
Nothing like this should be world readable. If it is installed from open source and read only, yes, that's fine, and necessary for NERSC.